R/rdeephaven/tests/testthat/test_table_handle_wrapper.R

```diff
@@ -0,0 +1,173 @@
+library(testthat)
+library(rdeephaven)
+
+setup <- function() {
+
+    df1 <- data.frame(string_col = c("I", "am", "a", "string", "column"),
+                int_col    = c(0, 1, 2, 3, 4),
+                dbl_col    = c(1.65, 3.1234, 100000.5, 543.234567, 0.00))
+
+    df2 <- data.frame(col1 = rep(3.14, 100),
+                        col2 = rep("hello!", 100),
+                        col3 = rnorm(100))
+
+    df3 <- data.frame(matrix(rnorm(10 * 1000), nrow = 10))
+
+    df4 <- data.frame(time_col = seq.POSIXt(as.POSIXct(Sys.Date()), as.POSIXct(Sys.Date()+30), by = "1 sec")[250000],
+                    bool_col = sample(c(TRUE, FALSE), 250000, TRUE), 
+                    int_col = sample(0:10000, 250000, TRUE))
+
+    # in order to test TableHandle, we need to have tables on the server that we know everything about.
+    # thus, we have to push these created tables to the server and get TableHandles to each of them.
+    # thus, we depend on the correctness of client$import_table(), ClientOptions$new(), and Client$new()
+    
+    # set up client
+    client_options <- ClientOptions$new()
+    client <- Client$new(target="localhost:10000", client_options=client_options)
+
+    # move dataframes to server and get TableHandles for testing
+    th1 <- client$import_table(df1)
+    th2 <- client$import_table(df2)
+    th3 <- client$import_table(df3)
+    th4 <- client$import_table(df4)
+
+    return(list("client" = client,
+                "df1" = df1, "df2" = df2, "df3" = df3, "df4" = df4,
+                "th1" = th1, "th2" = th2, "th3" = th3, "th4" = th4))
+}
+
+##### TESTING GOOD INPUTS #####
+
+test_that("is_static returns the correct value", {
+    data <- setup()
+
+    expect_true(data$th1$is_static())
+    expect_true(data$th2$is_static())
+    expect_true(data$th3$is_static())
+    expect_true(data$th4$is_static())
+
+    # TODO: test ticking tables when they can be created from R
+})
+
+test_that("nrow returns the correct number of rows", {
+    data <- setup()
+
+    expect_equal(data$th1$nrow(), nrow(data$df1))
+    expect_equal(data$th2$nrow(), nrow(data$df2))
+    expect_equal(data$th3$nrow(), nrow(data$df3))
+    expect_equal(data$th4$nrow(), nrow(data$df4))
+
+    # TODO: test nrow(data$th) when it is implemented
```

Comment:
todo.  

This looks like it is already tested.

Reply:
That seems like 10 lines of code.

Reply:
Ticket: https://github.com/deephaven/deephaven-core/issues/4090